### PR TITLE
JDK-8300222: Replace NULL with nullptr in share/logging

### PR DESCRIPTION
**Pull request by jdksjolen**
Do the conversion in the share/logging sub-directory and all of its files.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8300222](https://bugs.openjdk.org/browse/JDK-8300222): Replace NULL with nullptr in share/logging

### Reviewers
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12014/head:pull/12014` \
`$ git checkout pull/12014`

Update a local copy of the PR: \
`$ git checkout pull/12014` \
`$ git pull https://git.openjdk.org/jdk pull/12014/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12014`

View PR using the GUI difftool: \
`$ git pr show -t 12014`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12014.diff">https://git.openjdk.org/jdk/pull/12014.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/logging/logDecorations.cpp [hunk 1]
```diff
@@ -30,17 +30,17 @@
...
 const char* LogDecorations::host_name() {
   const char* host_name = Atomic::load_acquire(&_host_name);
-  if (host_name == NULL) {
+  if (host_name == nullptr) {
     char buffer[1024];
     if (os::get_host_name(buffer, sizeof(buffer))) {
       host_name = os::strdup_check_oom(buffer);
-      const char* old_value = Atomic::cmpxchg(&_host_name, (const char*)NULL, host_name);
-      if (old_value != NULL) {
+      const char* old_value = Atomic::cmpxchg(&_host_name, (const char*)nullptr, host_name);
```
Comment (coleenp) [inline src/hotspot/share/logging/logDecorations.cpp:42]: I wonder if you need this cast with nullptr ?

**Review by coleenp — APPROVED** [review]
Scrolled through can confirm only trivial change.  I had one question.

File: src/hotspot/share/logging/logAsyncWriter.cpp [hunk 2]
```diff
@@ -192,10 +192,10 @@ void AsyncLogWriter::initialize() {
   AsyncLogWriter* self = new AsyncLogWriter();
   if (self->_initialized) {
     Atomic::release_store_fence(&AsyncLogWriter::_instance, self);
-    // All readers of _instance after the fence see non-NULL.
+    // All readers of _instance after the fence see non-nullptr.
```
Comment (dholmes-ora) [inline src/hotspot/share/logging/logAsyncWriter.cpp:195]: Nit: When "null" or "NULL" are used in prose there is some ambiguity about whether  it is referring to the general concept of nullness or the value NULL. In this case I would have written this as "see non-null". I find "non-nullptr" a bit jarring.

**Review by dholmes-ora — APPROVED** [review]
Looks good. Thanks.

File: src/hotspot/share/logging/logDecorations.cpp (hunk 1 above)
Comment (jdksjolen) [reply src/hotspot/share/logging/logDecorations.cpp:42]: The nullptr argument is templated, so we do need the cast.

Comment (coleenp) [reply src/hotspot/share/logging/logDecorations.cpp:42]: Thanks for checking.  I won't ask on the next one. :)

Comment (kimbarrett) [reply src/hotspot/share/logging/logDecorations.cpp:42]: I'd like to add overloads for std::nullptr_t someday.

**Pull request closed without merging**